Objects/longobject.c

```diff
@@ -5421,6 +5432,59 @@ int_bit_length_impl(PyObject *self)
     return NULL;
 }
 
+/*[clinic input]
+int.bit_count
+
+Number of ones in the binary representation of self.
+
+>>> bin(13)
+'0b1101'
+>>> (13).bit_count()
+3
+[clinic start generated code]*/
+
+static PyObject *
+int_bit_count_impl(PyObject *self)
+/*[clinic end generated code: output=2e571970daf1e5c3 input=a428900d3e39a606]*/
+{
+    Py_ssize_t ndigits, i, bit_count = 0;
+    PyLongObject *result, *x, *y;
+
+    assert(self != NULL);
+    assert(PyLong_Check(self));
+
+    ndigits = Py_ABS(Py_SIZE(self));
+
+    for (i = 0; i < ndigits && i < PY_SSIZE_T_MAX/PyLong_SHIFT; i++) {
+        bit_count += popcount_digit(((PyLongObject *)self)->ob_digit[i]);
+    }
+
+    result = (PyLongObject *)PyLong_FromSsize_t(bit_count);
+    if (result == NULL) {
+        return NULL;
+    }
+
+    /* Use Python integers if bit_count would overflow. */
+    for (; i < ndigits; i++) {
```

Comment:
It's hard to imagine how this code would ever be exercised in practice (except possibly on a machine with a 32-bit `Py_ssize_t`), but just for fun I replaced the `i < PY_SSIZE_T_MAX/PyLong_SHIFT` condition in the previous `for` loop with `i < 0` and confirmed that with that change everything still passed, with no reference leaks.

Reply:
I based this on `long_bit_length`, which has similar overflow handling. Unfortunately it's not really feasible to exercise this branch in tests, so I also had to edit the condition to test it.